test/test_mma.cpp

```diff
@@ -163,4 +162,334 @@ INSTANTIATE_TEST_SUITE_P(
+  bool transpose_b = (layout == MmaLayout::TN || layout == MmaLayout::NN);
+
+  std::vector<int64_t> A_shape{getM(macro), getK(macro)},
+      B_shape{getK(macro), getN(macro)};
+
+  if (transpose_a) {
+    std::swap(A_shape[0], A_shape[1]);
+  }
+
+  if (transpose_b) {
+    std::swap(B_shape[0], B_shape[1]);
+  }
+
+  auto tv0 = makeConcreteTensor(A_shape, dtype);
+  auto tv1 = makeConcreteTensor(B_shape, dtype);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  TensorView* tv0b = nullptr;
+  TensorView* tv1b = nullptr;
+  int axes = 0;
+  switch (layout) {
+    case MmaLayout::TT:
+      // [M, K, N]
+      tv0b = broadcast(tv0, {false, false, true});
+      tv0b->reorder({{-1, -3}});
+      tv1b = broadcast(tv1, {true, false, false});
+      axes = 1;
+      break;
+    case MmaLayout::TN:
+      // [M, N, K]
+      tv0b = broadcast(tv0, {false, true, false});
+      tv0b->reorder({{-2, -3}});
+      tv1b = broadcast(tv1, {true, false, false});
+      axes = 2;
+      break;
+    case MmaLayout::NT:
+      // [K, M, N]
+      tv0b = broadcast(tv0, {false, false, true});
+      tv0b->reorder({{-1, -3}});
+      tv1b = broadcast(tv1, {false, true, false});
+      tv1b->reorder({{-2, -3}});
+      axes = 0;
+      break;
+    case MmaLayout::NN:
+      // [N, K, M]
+      tv0b = broadcast(tv0, {true, false, false});
+      tv1b = broadcast(tv1, {false, false, true});
+      tv1b->reorder({{-1, -3}});
+      axes = 1;
+      break;
+    default:
+      NVF_ERROR("Invalid layout");
+  }
+
+  auto tv2 = fusedMultiplySum(tv0b, tv1b, {axes});
+
+  switch (layout) {
+    case MmaLayout::TT:
+      // [M, K, N] -> [M, N, K]
+      tv2->reorder({{-2, -1}});
+      break;
+    case MmaLayout::TN:
+      // [M, N, K]
+      break;
+    case MmaLayout::NT:
+      // [K, M, N] -> [M, N, K]
+      tv2->reorder({{-3, -1}});
+      break;
+    case MmaLayout::NN:
+      // [N, K, M] -> [M, N, K]
+      tv2->reorder({{-1, -3}});
+      break;
+    default:
+      NVF_ERROR("Invalid layout");
+  }
+  tv2->commitLeafToRFactor();
+
+  fusion.addOutput(tv2);
+
+  auto mma_ops = ir_utils::getOpsOfType<MmaOp>(&fusion);
+  NVF_CHECK(
+      1 == mma_ops.size(),
+      "Invalid number of MmaOp instances in fusion definition, expected 1, got ",
+      mma_ops.size());
+  mma_ops.front()->setMacro(macro);
+
+  auto tv2c = tv2->cacheBefore();
+
+  tv0b->setMemoryType(MemoryType::Shared);
+  // Hopper tensor core assumes K major, so we are using !transpose_a here.
+  tv0b->applyMmaSwizzle(swizzle, !transpose_a);
+  tv1b->setMemoryType(MemoryType::Shared);
+  tv1b->applyMmaSwizzle(swizzle, transpose_b);
+
+  if (transpose_a) {
+    // TODO: Why do we need to transpose B if A is transposed? I don't really
+    // understand why, but empirically it works...
+    auto alloc = tv1b->getAllocationDomain();
+    std::swap(alloc[alloc.size() - 1], alloc[alloc.size() - 2]);
+    tv1b->setAllocationDomain(alloc, true);
+  }
```

Comment:
Maybe it makes more sense to put this section to `applyMmaSwizzle`, but because I don't understand why we need this, I will leave it here for now. In the future, I may move code around as I get deeper understanding of how tensor core works.